js/core/utils/ajax.js

```diff
@@ -2,8 +2,56 @@
 
 var $ = require("../renderer");
 
+var isStatusSuccess = function(status) {
+    return 200 <= status && status < 300;
+};
+
 var sendRequest = function(options) {
-    return $.ajax.call($, options);
+    options.xhrFields = options.xhrFields || {};
+    options.xhrFields.responseType = options.responseType;
+
+    if(options.upload) {
+        var xhr = new XMLHttpRequest();
+        var d = $.Deferred();
+        var result = d.promise();
+
+        xhr.open(options.method, options.url, true);
+        xhr.upload["onprogress"] = options.upload["onprogress"];
```

Comment:
The "onprogress" is not valid word for the linter. I am not sure it is correct to add the linter exception for this words.